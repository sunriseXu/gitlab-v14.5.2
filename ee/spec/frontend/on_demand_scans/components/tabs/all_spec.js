import { shallowMount } from '@vue/test-utils';
import AllTab from 'ee/on_demand_scans/components/tabs/all.vue';
import BaseTab from 'ee/on_demand_scans/components/tabs/base_tab.vue';

describe('AllTab', () => {
  let wrapper;

  // Finders
  const findBaseTab = () => wrapper.findComponent(BaseTab);

  const createComponent = (propsData) => {
    wrapper = shallowMount(AllTab, {
      propsData,
    });
  };

  beforeEach(() => {
    createComponent({
      itemsCount: 12,
    });
  });

  it('renders the base tab with the correct props', () => {
    expect(findBaseTab().props('title')).toBe('All');
    expect(findBaseTab().props('itemsCount')).toBe(12);
    expect(findBaseTab().props('fields')).toMatchSnapshot();
  });
});
