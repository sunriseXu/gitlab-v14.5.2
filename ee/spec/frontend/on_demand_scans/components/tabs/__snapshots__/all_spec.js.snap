// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllTab renders the base tab with the correct props 1`] = `
Array [
  Object {
    "columnClass": "gl-w-15",
    "key": "detailedStatus",
    "label": "Status",
  },
  Object {
    "key": "dastProfile.name",
    "label": "Name",
  },
  Object {
    "columnClass": "gl-w-13",
    "key": "scanType",
    "label": "Scan type",
  },
  Object {
    "key": "dastProfile.dastSiteProfile.targetUrl",
    "label": "Target",
  },
  Object {
    "columnClass": "gl-w-15",
    "key": "createdAt",
    "label": "Start date",
  },
  Object {
    "columnClass": "gl-w-13",
    "key": "id",
    "label": "Pipeline",
  },
]
`;
